app/src/main/kotlin/edu/stanford/bdh/engagehf/health/time/TimePickerStateMapper.kt

```diff
@@ -40,15 +40,18 @@ class TimePickerStateMapper @Inject constructor(
         timePickerState: TimePickerState,
     ) = timePickerState.copy(
         selectedDate = date,
-        selectedDateFormatted = format(date)
+        selectedDateFormatted = dateFormatter.formatUTC(date, DATE_FORMAT)
     )
 
     fun mapInstant(timePickerState: TimePickerState): Instant = with(timePickerState) {
-        val zoneId = ZoneId.of("UTC")
-        val date = selectedDate.atZone(zoneId).toLocalDate()
-        return LocalDateTime.of(date, selectedTime).atZone(zoneId).toInstant()
+        val date = selectedDate.atZone(UTC_ZONE_ID).toLocalDate()
```

Comment:
Minor: Perhaps it would be useful to rename `selectedDate` to `selectedInstant` after all the changes to better reflect the actual type. This would make this line, for example, even easier to understand.